cmd/core/pfcp_connection.go

```diff
@@ -101,3 +114,42 @@ func (connection *PfcpConnection) SendMessage(msg message.Message, addr *net.UDP
 	}
 	return nil
 }
+
+func (connection *PfcpConnection) ProcessHeadlessAssociations() {
+	for assocAddr, assoc := range connection.NodeAssociations {
+		if assoc.IsExpired() {
+			log.Printf("Pruning expired node association: %s", assocAddr)
+			connection.DeleteAssociation(assocAddr)
+		}
+		if assoc.cancelRetries == nil {
```

Comment:
Looks strange. Seems like should be called only if association is not expired. I'd proposed extract method like `assoc.IsHeartbeatScheduled`